integration_tests/tests/transformation-tests.ts

```diff
@@ -0,0 +1,46 @@
+import path from "path";
+import fs from "fs";
+import { resourcePrefix } from "../../src/handlers/int-test-support/helpers/envHelper";
+import {
+  S3Object,
+  putS3Object,
+  listS3Objects,
+  deleteS3FolderBasedOnDate,
+} from "../../src/handlers/int-test-support/helpers/s3Helper";
+import { wait } from "../../src/handlers/int-test-support/helpers/commonHelpers";
+
+const prefix = resourcePrefix();
+const storageBucket = `${resourcePrefix()}-storage`;
+const folderPrefix = "btm_transactions";
+
+describe("\n Upload verify fake csv to transformation bucket tests and check valid events from csv match with s3 events", () => {
+  beforeAll(async () => {
+    // before the test runs deletes the btm_transaction/datePrefix folder created by previous run
+    await deleteS3FolderBasedOnDate(storageBucket, folderPrefix);
+    console.log("Existing sub folders within btm_transactions are deleted");
+    const fakeCsvFile = "../payloads/fakeBillingReport.csv";
+    const filePath = path.join(__dirname, fakeCsvFile);
+    const csvFileData = fs.readFileSync(filePath);
+    const testObject: S3Object = {
+      bucket: `${prefix}-transformation`,
+      key: `fakeBillingReport.csv`,
+    };
+
+    await putS3Object({ target: testObject, data: csvFileData });
+    await wait(10000);
```

Comment:
What is this unconditionally waiting for?

Reply:
I believe this was because of the need to wait for the csv file that had been put in the transformation bucket to get transformed to valid events that are then put in the storage/btm_transactions/yyyy-mm-dd bucket.